### PR TITLE
Adds tests for connectivity resolution

### PR DESCRIPTION
**Pull request by johnnyman727**
Intended to supplement #445.

### CODE REVIEW

File: lib/tessel/wifi.js [hunk 1]
```diff
@@ -118,7 +118,7 @@ Tessel.prototype.connectToNetwork = function(opts) {
                     };
                     // End the connection
                     remoteProcess.close();
-                  }, 10000);
+                  }, Tessel.__wifiConnectionTimeout);
```
Comment (rwaldron) [inline lib/tessel/wifi.js]: `__` is unnecessary.

File: test/common/remote-process-simulator.js [hunk 2]
```diff
@@ -33,6 +33,10 @@ function RemoteProcessSimulator() {
     // Keep recording what gets written
     self.stdin._write = stdinRecorder;
   });
+
+  this.close = function() {
+    this.emit('close');
+  }.bind(this);
```
Comment (rwaldron) [inline test/common/remote-process-simulator.js]:
```js
this.close = () => this.emit('close');
```

;)

Comment (johnnyman727) [inline test/common/remote-process-simulator.js]: Gah it's so hard for this old dog to learn new tricks.

**Comment by johnnyman727** [comment]
I believe @HipsterBrown is going to write better tests for his branch and close this one but I updated with @rwaldron's feedback just in case.

File: lib/tessel/wifi.js [hunk 3]
```diff
@@ -152,6 +152,10 @@ Tessel.prototype.connectToNetwork = function(opts) {
   });
 };
 
+// How long Tessel will wait for a status after attempting to connect
+// Exposed for easier testing
+Tessel._wifiConnectionTimeout = 10000;
```
Comment (rwaldron) [inline lib/tessel/wifi.js]: I'm not sure why the `_` prefix is necessary at all?

Comment (johnnyman727) [inline lib/tessel/wifi.js]: I was just indicating that this var is used internally.

**Comment by johnnyman727** [comment]
Replaced by #471.

**Pull request closed without merging**